File: 28-implement-strstr/28-implement-strstr.go
func strStr(haystack string, needle string) int {
	for i := 0; i < len(haystack); i++ {
		if haystack[i] == needle[0] {
			if len(haystack) >= i+len(needle) {
				if haystack[i:i+len(needle)] == needle {
					return i
				}
			} else {
				return -1
			}
		}
	}
	return -1
}

File: 28-implement-strstr/28-implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  const first = needle[0]
  const len = needle.length

  for (let i = 0; i < haystack.length; i++) {
    if (haystack[i] === first) {
      if (haystack.substring(i, i + len) === needle) {
        return i;
      }
    }
  }
  return -1
};